tests/unit/specification/parser_test.py

```diff
@@ -1925,6 +1925,203 @@ def test_parse_error_invalid_context_clause(tmp_path: Path) -> None:
+                  Final => Terminated
+               is
+                  Message : M;
+               begin
+                  state Start is
+                  begin
+                     Channel'Read (Message);
+                  transition
+                     goto Reply
+                        if Message'Valid
+                     goto Terminated
+                  end Start;
+
+                  state Reply is
+                  begin
+                     Channel'Write (Message);
+                  transition
+                     goto Terminated
+                  end Reply;
+
+                  state Terminated is null state;
+               end Session;
+
+            end Test;
+        """,
+        r'^<stdin>:11:62: parser: error: duplicate aspect "Writable"',
+    )
+
+
+def test_parse_error_duplicate_checksum_aspect() -> None:
+    assert_error_string(
+        """
+            package Test is
+               type T is mod 2**8;
+               type M is
+                  message
+                     F1 : T;
+                     F2 : T;
+                     C1 : T;
+                     F3 : T;
+                     C2 : T
+                        then F4
+                           if C1'Valid_Checksum and C2'Valid_Checksum;
+                     F4 : T;
+                  end message
+                     with Checksum => (C1 => (F3, F1'First .. F2'Last),
+                                       C2 => (C1'Last + 1 .. C2'First - 1)),
+                          Checksum => (C1 => (F3, F1'First .. F2'Last),
+                                       C2 => (C1'Last + 1 .. C2'First - 1)),
+                          Byte_Order => Low_Order_First;
+            end Test;
+        """,
+        # ISSUE: Componolit/RecordFlux-parser#56
+        # Delete the following line when ticket mentioned above is solved.
+        "^<stdin>:17:27: parser: error: Expected 'Byte_Order', got 'First'",
+        # Uncomment this when ticket #56 is solved.
+        #        r'^<stdin>:17:27: parser: error: duplicate aspect "Checksum"',
+    )
+
+
+def test_parse_error_duplicate_byte_order_aspect() -> None:
+    assert_error_string(
+        """
+            package Test is
+               type T is mod 2**8;
+               type M is
+                  message
+                     F1 : T;
+                     F2 : T;
+                     C1 : T;
+                     F3 : T;
+                     C2 : T
+                        then F4
+                           if C1'Valid_Checksum and C2'Valid_Checksum;
+                     F4 : T;
+                  end message
+                     with Checksum => (C1 => (F3, F1'First .. F2'Last),
+                                       C2 => (C1'Last + 1 .. C2'First - 1)),
+                          Byte_Order => Low_Order_First,
+                          Byte_Order => Low_Order_First;
+            end Test;
+        """,
+        r'^<stdin>:18:27: parser: error: duplicate aspect "ByteOrderAspect"',
```

Comment:
That's what the error message generates.

Reply:
That means changing aspect.kind_name

Reply:
In that case I think the error message should stay the same to keep the code consistent then.

Reply:
The error messages are intended for users of the tool. Having a error message mentioning "ByteOrderAspect", while the aspect is actually called "Byte_Order" in the specification, is confusing. Keeping the code as it as, just because it is easier, is no option here.

Reply:
First, I don't think ByteOrderAspect is confusing, it literally literally has Byte Order and Aspect in it. and I already proposed a solution to change kind_name if you don't like the exact wording. I don't see another way to do it.